driver-mapping/src/main/java/com/datastax/driver/mapping/ReflectionUtils.java

```diff
@@ -25,12 +26,19 @@
 import java.lang.reflect.*;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
 
 /**
  * Utility methods related to reflection.
  */
 class ReflectionUtils {
 
+    private static final Set<String> EXCLUDED_PROPERTIES = ImmutableSet.of(
```

Comment:
I would actually lean towards the pre-3.1.0 behaviour and a opt-in approach for having the mapper scanning all fields in subclasses like the Groovy's metaclass and so on, via indeed something like `MappingConfiguration` if that's possible. Because if we start making a special case for the Groovy metaclass we probably will end up having to do other special cases over and over later.